cmd/service_linux.go

```diff
@@ -0,0 +1,316 @@
+*/
+
+package cmd
+
+import (
+	"bufio"
+	"errors"
+	"fmt"
+	"io/fs"
+	"os"
+	"os/exec"
+	"os/user"
+	"strings"
+
+	"github.com/Seagate/cloudfuse/common"
+	"github.com/spf13/cobra"
+)
+
+type serviceOptions struct {
+	ConfigFile string
+	MountPath  string
+}
+
+var servOpts serviceOptions
+
+// Section defining all the command that we have in secure feature
+var serviceCmd = &cobra.Command{
+	Use:               "service",
+	Short:             "Manage cloudfuse startup process on Linux",
+	Long:              "Manage cloudfuse startup process on Linux",
+	SuggestFor:        []string{"ser", "serv"},
+	Example:           "cloudfuse service install",
+	FlagErrorHandling: cobra.ExitOnError,
+	RunE: func(cmd *cobra.Command, args []string) error {
+		return errors.New("missing command options\n\nDid you mean this?\n\tcloudfuse service install\n\nRun 'cloudfuse service --help' for usage")
+	},
+}
+
+var installCmd = &cobra.Command{
+	Use:               "install [mount path] [config path]",
+	Short:             "Installs a service file for a single mount with Cloudfuse. Requires elevated permissions.",
+	Long:              "Installs a service file for a single mount with Cloudfuse which remounts any active previously active mounts on startup. elevated permissions.",
+	SuggestFor:        []string{"ins", "inst"},
+	Example:           "cloudfuse service install",
+	FlagErrorHandling: cobra.ExitOnError,
+	Args:              cobra.ExactArgs(2),
+	RunE: func(cmd *cobra.Command, args []string) error {
+
+		// 1. get the cloudfuse.service file from the setup folder and collect relevant data (user, mount, config)
+
+		// get current dir
+		dir, err := os.Getwd()
+		if err != nil {
+			return fmt.Errorf("error: [%s]", err.Error())
+		}
+
+		// TODO: get arguments of cloudfuse service install and pass that to be values written to the service file's config file and mount point paths.
+		mountPath := args[0]
+		configPath := args[1]
+
+		mountExists := common.DirectoryExists(mountPath)
+		if !mountExists {
+			return fmt.Errorf("error, the mount path provided does not exist")
+			// TODO: add useage output upon failure with input
+		}
+		isDirEmpty := common.IsDirectoryEmpty(mountPath)
+		if !isDirEmpty {
+			return fmt.Errorf("error, the mount path provided is not empty")
+			// TODO: add useage output upon failure with input
+		}
+
+		_, err = os.Stat(configPath)
+		if errors.Is(err, fs.ErrNotExist) {
+			return fmt.Errorf("error, the configfile path provided does not exist") // TODO: add useage output upon failure with input
+		}
+
+		serviceFile, err := newSericeFile(mountPath, configPath, dir)
+		if err != nil {
+			return fmt.Errorf("error when attempting to create service file: [%s]", err.Error())
+		}
+
+		// 2. retrieve the newUser account from cloudfuse.service file and create it if it doesn't exist
+
+		// assumes dir is in cloudfuse repo dir
+		serviceData, err := collectServiceData(fmt.Sprintf("%s/setup/cloudfuse.service", dir))
```

Comment:
You assume that this directory exists. But if the user just runs the installer for cloudfuse this file will not exist, you will need to write it for them.

Reply:
Are you suggesting I should have a hard coded template to reference? or is there another way to have a template I can always reference?

Reply:
Yeah, you could just use a template either in the form of a string that you add the mountpoint and config too or something like text/template. Possibly something like this https://github.com/pioz/god.